### ISSUE TITLE
Discussion and Q&A of Outreachy winter 2022 project

### ISSUE THREAD

**Issue by cesswairimu** [publiclab/Leaflet.DistortableImage#1130]
Hi all :wave: 
This is a discussion issue for the "MapKnitter Lite: fetch images from Internet Archive" project. The original idea can be found [here](https://publiclab.org/wiki/gsoc-ideas#MapKnitter+Local) and the project progress [here](https://github.com/publiclab/Leaflet.DistortableImage/issues/998)  _feel free to use this issue for reference but please lets keep the outreachy project questions on this issue as #998 is a planning issue_.

Thanks so much for your interest! and we are excited to have a chance to work with you :heart: :heart: . 

#### Tasks: 
- When opening a map, detect a key like "texas-barnraising" from a longer URL like https://archive.org/details/texas-barnraising
- Add a geocoding search box (where you type a location and it moves the map view) to the demo so upon creating a map users can navigate to a specific place
- Add a “drag image onto page to add” feature to the Leaflet.DistortableImage demo so people can import multiple images into maps in the demo at https://publiclab.github.io/Leaflet.DistortableImage/examples/export.html
- Allow current map state (like a "save game" button) to be exported to a downloadable “mapknitter-saved.json” format (see https://github.com/publiclab/image-sequencer-app#usage for possible JSON format and https://gis.stackexchange.com/q/298808 / https://geojson.org/ for official GeoJSON standard)
- Add local saving of maps, via localStorage (may be limited by filesize?) so you don’t lose your work when you refresh the page (using same format as “mapknitter-saved.json”)

#### Stretch goals:
- local full-res map exporting: 3 possible approaches:
- switch to Ruby exporter in exporting demo: https://github.com/publiclab/Leaflet.DistortableImage/issues/751 (maybe easiest, although it means you’d have to find how to send the local images to the remote exporter)
- rework the JS exporter for local use: https://github.com/publiclab/image-sequencer-app (challenging to rework server-side multi-threaded Node code)
- or use the existing webgl-distort individual image export (https://github.com/publiclab/Leaflet.DistortableImage/#full-resolution-download) and stack them into one giant image using https://github.com/nodeca/pica (hard but relatively straightforward)

**Comment by 7malikk** [comment]
Hello @jywarren, (@cesswairimu @TildaDares)  I'm an Outreachy applicant, here to help improve Mapknitter.
I'm a frontend web developer skilled in Javascript and Reactjs among others, I noticed there are some tasks under the sub-topic: "Smaller Issues", I'd like to give them a go for my FTO. This task is from the planning issue [998](https://github.com/publiclab/Leaflet.DistortableImage/issues/998)

**Comment by Dhei-vid** [comment]
I am also an outreachy applicant and want to contribute to the project. I am a frontend developer (React JS) but this I am a newbie to open source contribution and I would appreciate if I am guided.

**Comment by yukiyuli** [comment]
Hello everyone, I am an Outreachy applicant. I am interested in this project. I have the knowledge of HTML, CSS, JavaScript and Vue. Please advise what the next step I need to do.

**Comment by 7malikk** [comment]
@yukiyuli Hello there, welcome to the community, assuming you've gone through the necessary docs, you can start by requesting to be assigned to an issue [here](https://github.com/publiclab/plots2/issues/11105), while you get familiar with the codebase. Cheers

**Comment by 7malikk** [comment]
Hello again, @cesswairimu I'm also interested in [998](https://github.com/publiclab/Leaflet.DistortableImage/issues/998), with the resource @jywarren linked to the geocoding search box task, I checked through I'd like to suggest [this plugin](https://smeijer.github.io/leaflet-geosearch/), for that task, with some guidance I would also like to work on the task.

**Comment by cesswairimu** [comment]
Hi all, info on the next steps can be found #1131. Thanks 

@7malikk the unchecked tasks on #998 are the tasks that most will be worked on during the internship period. The task you mentioned on plugin, it may be a bit of a discussion before we can settle on a plugin / an approach. You could go through the other tasks though and if you can see a sub-task that can be worked on independently and is straight-forward, you can open a new issue for the subtask  you are free to give it a try/ start a discussion. Thanks

**Comment by 7malikk** [comment]
@cesswairimu Oh, alright thanks a lot for the feedback, I'll check for a straightforward task.

**Comment by leilayesufu** [comment]
Hi @cesswairimu 
I’ve gone through the issue tasks and they’re quite interesting and compelling 
I think the first task can be achieved using regex to detect keys in the url 
The second seems the easiest to me, to create a search box and style it and use the input value on click or on key enter to navigate the map
The third task of drag image onto page to add got me wondering what if user is using a mobile device to access the map
The fourth tasks seems similar to the 2nd one with a download to download the map’s nodes position and cm_per_pixel in a JSON format 
The last task with the local storage, I think an issue we’d face is what happens if a users local storage becomes full?

**Comment by jywarren** [comment]
Thank you for your ideas @leilayesufu -- regex is a good idea. Should there be a whole function to parse out keys in the URL?

**Comment by sainamercy** [comment]
@jywarren @cesswairimu regarding this [comment](https://github.com/publiclab/Leaflet.DistortableImage/issues/1130#issuecomment-1279796153). What is the goal of this task `When opening a map, detect a key like "texas-barnraising" from a longer URL like https://archive.org/details/texas-barnraising`
